app/_components/_sections/SavedWorkoutsList.tsx

```diff
@@ -0,0 +1,196 @@
+
+interface SavedWorkoutsModalProps {
+  workouts: Workout[];
+  onClose: () => void;
+  refetchSavedWorkouts: () => void;
+}
+
+const calculateTotalDuration = (workout: Workout) => {
+  let totalDurationSeconds = 0;
+
+  workout.workout_schedule?.forEach((schedule) => {
+    schedule.exercises?.forEach((exercise) => {
+      const sets = exercise.sets || 0;
+      const restSeconds = exercise.rest_seconds || 0;
+      const setDurationSeconds = 120;
+
+      const exerciseDurationSeconds = sets * setDurationSeconds;
+      const totalRestBetweenSets = (sets - 1) * restSeconds;
+
+      totalDurationSeconds += exerciseDurationSeconds + totalRestBetweenSets;
+    });
+  });
+
+  return Math.floor(totalDurationSeconds / 60);
+};
+
+const SavedWorkoutsModal: React.FC<SavedWorkoutsModalProps> = ({ workouts, onClose, refetchSavedWorkouts }) => {
+  const { t } = useTranslation("global");
+  const { mutate: deleteWorkout } = useDeleteWorkout();
+  const [modalOpen, setModalOpen] = useState(false);
+  const [selectedWorkout, setSelectedWorkout] = useState<Workout | null>(null);
+  const [clickedStates, setClickedStates] = useState<{ [key: string]: boolean }>({});
+  const [workoutList, setWorkoutList] = useState<Workout[]>(workouts);
+  const [loadingWorkoutId, setLoadingWorkoutId] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (workouts) {
+      setWorkoutList(workouts);
+      const initialClickedStates = workouts.reduce((acc, workout) => {
+        acc[workout._id] = true;
+        return acc;
+      }, {} as { [key: string]: boolean });
+      setClickedStates(initialClickedStates);
+    }
+  }, [workouts]);
+
+  const handleBookmarkClick = (workout: Workout) => {
+    const duration = calculateTotalDuration(workout);
+    setSelectedWorkout({ ...workout, duration });
+    setModalOpen(true);
+  };
+
+  const handleConfirmDelete = () => {
+    if (selectedWorkout) {
+      deleteWorkout(
+        { queryParams: { workout_id: selectedWorkout._id } },
+        {
+          onSuccess: () => {
+            setClickedStates((prev) => {
+              const updatedStates = { ...prev };
+              delete updatedStates[selectedWorkout._id];
+              return updatedStates;
+            });
+
+            setWorkoutList((prev) => prev.filter(workout => workout._id !== selectedWorkout._id));
+
+            setSelectedWorkout(null);
+            setModalOpen(false);
+
+            refetchSavedWorkouts();
+          },
+          onError: (error) => {
+            console.error("Error deleting workout:", error);
+          },
+        }
+      );
+    }
+  };
+
+  const handleCancelDelete = () => {
+    setModalOpen(false);
+    setSelectedWorkout(null);
+  };
+
+  const handleWorkoutClick = (e: React.MouseEvent<HTMLDivElement>, workoutId: string) => {
+    e.preventDefault();
+    if (loadingWorkoutId) return;
+    setLoadingWorkoutId(workoutId);
+
+    setTimeout(() => {
+      window.location.href = `/workouts/plan/${workoutId}`;
```

Comment:
Dont use this to redirect, please use

router.push(`/workouts/plan/${workoutId}`);